lib/node_modules/@stdlib/ndarray/base/unary/test/test.2d.js

```diff
@@ -0,0 +1,1175 @@
+
+	expected = new Complex128Array([
+		10.0,
+		20.0,
+		30.0,
+		40.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		90.0,
+		100.0,
+		110.0,
+		120.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0
+	]);
+	t.strictEqual( isSameComplex128Array( y.data, expected ), true, 'returns expected value' );
+	t.end();
+
+	function scale( z ) {
+		return new Complex128( real(z)*10.0, imag(z)*10.0 );
+	}
+});
+
+tape( 'the function applies a unary callback to each indexed element of a 2-dimensional ndarray (column-major, non-contiguous, mixed sign strides, accessors)', function test( t ) {
+	var expected;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+	var y;
+
+	dt = 'complex128';
+	ord = 'column-major';
+	sh = [ 2, 2 ];
+	st = [ 1, -4 ];
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, new Complex128Array( oneTo( numel( sh ) * 4, 'float64' ) ), sh, st, o, ord );
+	y = ndarray( dt, zeros( 8, dt ), sh, st, o, ord );
+
+	unary( [ x, y ], scale );
+
+	expected = new Complex128Array([
+		10.0,
+		20.0,
+		30.0,
+		40.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0,
+		90.0,
+		100.0,
+		110.0,
+		120.0,
+		0.0,
+		0.0,
+		0.0,
+		0.0
+	]);
+	t.strictEqual( isSameComplex128Array( y.data, expected ), true, 'returns expected value' );
+	t.end();
+
+	function scale( z ) {
+		return new Complex128( real(z)*10.0, imag(z)*10.0 );
+	}
+});
+
+tape( 'the function applies a unary callback to each indexed element of a 2-dimensional ndarray (column-major, non-contiguous, large arrays, accessors)', function test( t ) {
+	var expected;
+	var bsize;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+	var y;
+
+	dt = 'complex128';
+	ord = 'column-major';
+
+	bsize = blockSize( dt );
+	sh = [ bsize*2, 2 ];
+	st = [ 2, -bsize*4 ];
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, new Complex128Array( ones( numel( sh ) * 4, 'float64' ) ), sh, st, o, ord );
+	y = ndarray( dt, zeros( numel( sh ) * 2, dt ), sh, st, o, ord );
+
+	unary( [ x, y ], scale );
+
+	expected = new Complex128Array( x.length*2 );
+	gfill( x.length, new Complex128( 10.0, 10.0 ), expected, st[ 0 ] );
+
+	t.strictEqual( isSameComplex128Array( y.data, expected ), true, 'returns expected value' );
+	t.end();
+
+	function scale( z ) {
+		return new Complex128( real(z)*10.0, imag(z)*10.0 );
+	}
+});
+
+tape( 'the function applies a unary callback to each indexed element of a 2-dimensional ndarray (column-major, non-contiguous, large arrays, accessors)', function test( t ) {
+	var expected;
+	var bsize;
+	var ord;
+	var sh;
+	var st;
+	var dt;
+	var o;
+	var x;
+	var y;
+
+	dt = 'complex128';
+	ord = 'column-major';
+
+	bsize = blockSize( dt );
+	sh = [ 2, bsize*2 ];
+	st = [ -2, 4 ];
+	o = strides2offset( sh, st );
+
+	x = ndarray( dt, new Complex128Array( ones( numel( sh ) * 4, 'float64' ) ), sh, st, o, ord );
+	y = ndarray( dt, zeros( numel( sh )*2, dt ), sh, st, o, ord );
+
+	unary( [ x, y ], scale );
+
+	expected = new Complex128Array( x.length*2 );
+	gfill( x.length, new Complex128( 10.0, 10.0 ), expected, st[ 0 ] );
```

Comment:
For future reference, we also have `@stdlib/blas/ext/base/zfill`, which is a package you added. :)

This works. It will just be a bit slower.